### PR TITLE
Refactor all code for security and new API version

### PR DESCRIPTION
**Pull request by dschach**
Resubmitted to allow for recent changes

I have updated all the code to latest API (with required tests being moved to new classes). Fixed unnecessary global modifiers, variables in the wrong place, unused variables, and more. This project is ready for SFDX now.
I deleted the src folder because it is unnecessary - people can install from the appropriate folder in force-app.
As this is unmanaged, deploying on top of existing (unmodified) code won't be an issue; users should delete unnecessary classes like Message, ReportFinderUtil, etc.

### CODE REVIEW

File: config/project-scratch-def.json [hunk 1]
```diff
@@ -1,5 +1,18 @@
 {
-  "orgName": "jrattanpal company",
+  "orgName": "dschach company",
```
Comment (dschach) [inline config/project-scratch-def.json:2]: This file probably doesn't need to be committed, but it's here.

File: force-app/main/default/classes/CSUtils.cls [hunk 2]
```diff
@@ -382,8 +384,26 @@ public with sharing class CSUtils {
 		email.setSubject(subject);
 		email.setPlainTextBody(message);
 		email.setHtmlBody(message);
+		if(canSendEmails()){
+			Messaging.sendEmail(new Messaging.SingleEmailMessage[] {email});
+		}
+	}
+
+	private static Boolean orgCanSendEmails = null;
```
Comment (dschach) [inline force-app/main/default/classes/CSUtils.cls:392]: Add a method to check if we can send an email so tests don't fail if sandbox email permission is disabled.

**Comment by jrattanpal** [comment]
Sounds good. I have pulled it locally. Let me test it too (2 sets of eyes) and will submit to master once ready.

Thanks again!  This is great

**Comment by jrattanpal** [comment]
@dschach 

I am changing base to another branch. I noticed some very minor thing which need to be fixed. I will fix those in a cleanup branch (after pulling your changes) And then merge to master.

Are you ok with it?

**Comment by dschach** [comment]
Sounds good. I'm curious what needs to be changed, but happy to take a look at your other branch once it's ready to merge back to master.

**Comment by jrattanpal** [comment]
> Sounds good. I'm curious what needs to be changed, but happy to take a look at your other branch once it's ready to merge back to master.

I'll let you know once I submit the changes. There were minor things but It maybe that I am not doing the resolve correctly.

for example: 
- reports/FOLDER.XML file was missing
- I received error that "viewSurveyResultsComponentController.cls" name didn't match with class name. Class name started with Capital V. I have never seen this error before but renaming the file to capital V worked. I am not sure why it worked for you but not me. Strange.

**Comment by jrattanpal** [comment]
These are the minor changes I had to make becuase of filename and class name mismatch. Not sure If those were caught because I am on Mac. I was finally able to push this in a scratch org.

https://github.com/SalesforceLabs/survey-force/commit/d6e0c5737b066546f9c05a78ed66bedc1f4689dd

**Comment by jrattanpal** [comment]
It seems you may not have pulled all changes from master where I renamed some classes this morning. That is causing conflicts. I will need to manually resolve those. It will take some time

**Comment by jrattanpal** [comment]
@dschach 

All your changes are in master. I had to do it from command line because of conflicts. Please check the master branch to make sure all is good. You can run orgInit to create new org (if you want) and test. 

please let me know.

I will close this pull request as it's no longer need.

THANK YOU again for this. This is VERY HELPFUL!

**Pull request closed without merging**